services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/application/ApplicationServiceITest.kt

```diff
@@ -597,6 +598,58 @@ class ApplicationServiceITest : DatabaseTest() {
             }
         }
 
+        @Test
+        fun `when application is in Allu should initialize access for new contacts`() {
+            val cableReport =
+                createCableReportApplicationData(
+                    areas = listOf(aleksanterinpatsas),
+                    customerWithContacts = hakijaCustomerContact,
+                    contractorWithContacts = suorittajaCustomerContact,
+                )
+            val application =
+                hankeService.generateHankeWithApplication(
+                    CableReportWithoutHanke(CABLE_REPORT, cableReport),
+                    USERNAME,
+                )
+            mockSendAllu(applicationRepository.findById(application.id!!).orElseThrow())
+            val capturedEmails = mutableListOf<ApplicationNotificationData>()
+            with(cableReportServiceAllu) {
+                justRun { update(21, any()) }
+                every { getApplicationInformation(any()) } returns createAlluApplicationResponse(21)
+                justRun { addAttachment(21, any()) }
+            }
+            with(emailSenderService) {
+                justRun { sendHankeInvitationEmail(any()) }
+                justRun { sendApplicationNotificationEmail(capture(capturedEmails)) }
+            }
+
+            applicationService.updateApplicationData(
+                application.id!!,
+                cableReport.copy(
+                    representativeWithContacts = asianHoitajaCustomerContact,
+                    propertyDeveloperWithContacts = rakennuttajaCustomerContact,
+                    contractorWithContacts =
+                        suorittajaCustomerContact.changeContactEmails("new.mail@foo.fi")
+                ),
+                USERNAME
+            )
+
+            assertThat(capturedEmails).hasSize(3)
+            assertThat(capturedEmails)
+                .areValid(application.applicationType, application.hankeTunnus)
```

Comment:
This could use something like:
```
            assertThat(capturedEmails)
                .extracting { it.recipientEmail }
                .containsExactlyInAnyOrder(
                    "new.mail@foo.fi",
                    asianHoitajaCustomerContact.contacts[0].email,
                    rakennuttajaCustomerContact.contacts[0].email
                )
```
So that:
1. There's a check that we send the right emails. `areValid` doesn't check we send one per email and it uses a list with an extra email we don't want to send to here.
2. It makes it a lot easier to read this test when one can clearly see which emails are supposed to be sent.

Reply:
Added. 

Also removed:  assertThat(capturedEmails).hasSize(3) as redundant.